### DISCUSSION TITLE
how to add the translation to the entity?

### DISCUSSION THREAD

**Discussion by KratorD** [zikula/core#5338] [General]
Hello all.
I'm using Zikula 3.0. I have the follow:
`$criatura = new CriaturaEntity();
            
            $criatura->setWorkflowState('initial');
            $criatura->setFactionId((int)$row['id_castillo']);
            $criatura->setName($row['nombreIng']);`

The field "Name" is translatable. How can I add the translated value to the entity?

Thanks.

**Comment by Guite** [comment]
Here is a basic example to follow: https://github.com/doctrine-extensions/DoctrineExtensions/blob/main/doc/translatable.md#basic-usage-examples

**Comment by KratorD** [comment]
Thanks, it's working


```
$repoTrans = $em->getRepository('TdM\HeroesIIModule\Entity\\CriaturaTranslationEntity');

$criatura = new CriaturaEntity();
$criatura->setWorkflowState('initial');
$criatura->setFactionId((int)$row['id_castillo']);
$criatura->setName($row['nombreIng']);
$repoTrans->translate($criatura, 'name', 'es', $row['nombre'])
$action = 'submit';
            try {
                $em->persist($criatura);
                $success = $workflowHelper->executeAction($criatura, $action);
            }
```

**Comment by KratorD** [comment]
Thanks for all Guite
